### ISSUE TITLE
ingress_application_gateway overwrites settings for existing application gateway

### ISSUE THREAD

**Issue by clbx** [hashicorp/terraform-provider-azurerm#18203]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.7

### AzureRM Provider Version

3.15.1

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster, azurerm_application_gateway

### Terraform Configuration Files

Kubernetes Cluster
```hcl



resource "azurerm_kubernetes_cluster" "k8s" {
  name                = var.name
  location            = var.location
  resource_group_name = var.rg
  dns_prefix          = var.dns_prefix
  kubernetes_version  = "1.23.5"

  node_resource_group = local.node_resource_group_name

  default_node_pool {
    name           = "agentpool"
    node_count     = 3
    vm_size        = "Standard_DS2_v2"
    vnet_subnet_id = azurerm_subnet.aks_subnet.id
  }

  identity {
    type = "SystemAssigned"
  }

  ingress_application_gateway {
    gateway_id = azurerm_application_gateway.appgw.id    
  }

  network_profile {
    load_balancer_sku = "standard"
    network_plugin    = "azure"
  }

}
```

Application Gateway
```hcl
resource "azurerm_application_gateway" "appgw" {
  name = local.appgw_name
  resource_group_name = azurerm_resource_group.resource_group.name
  location = var.location


  trusted_root_certificate {...}
  ssl_certificate {...}

  # App Gateway SKU - Determines what tier of product it is
  sku {
    name = "WAF_v2"
    tier = "WAF_v2"
    capacity = "1"
  }

  gateway_ip_configuration {...}
  frontend_port {...}
  frontend_ip_configuration {...}
  http_listener {...}
  request_routing_rule {...}
  backend_address_pool {...}
  backend_http_settings {...}
  probe {...}

}
```

### Debug Output/Panic Output

```shell
There is no unexpected output from Terraform or errors
```

### Expected Behaviour

The Kubernetes Cluster created by ``azurerm_kubernetes_cluster k8s`` module uses the referenced application gateway ``gateway_id`` in the ``ingress_application_gateway`` section unmodified. 

### Actual Behaviour

The kubernetes cluster resource completely reconfigures the application gateway to some kind of "defaults". All of the configuration done in the application gateway (``gateway_ip_configuration``, ``backend_address_pool``, ``backend_http_settings``, etc.) is completely overwritten by default values. This is observable in the application gateway log where all of the settings in the application gateway module are created and a few minutes later overwritten. 

It seems like when the aks cluster module does the work described in ``ingress_application_gateway`` it disregards any setting that is already there and overwrites the information. 

It looks like to me that the resources are stored as JSON, so for example the application gateway will write the configuration described in ``probes`` section of the ``azurerm_application_gateway`` resource to ``probes[0]``. And then the ``ingress_application_gateway`` section of the ``azurerm_kubernetes_cluster`` overwrites that with some kind of default settings. 

I attempted to get around this by making the application gateway resouce rely on the kubernetes module. This applied settings as expected, but the Ingress controller is expecting resources (http_listeners,address pools, etc. ) to be as they were with the default settings.

TLDR: When supplying an already existing application gateway to the kubernetes cluster using ``ingress_application_gateway`` It overwrites the configuration that already exists. 

### Steps to Reproduce

1. Create the application gateway and kubernetes cluster in the same terraform apply
2. View application gateway

### Important Factoids

N/A

### References

#15693 is tangentially related.

**Comment by clbx** [comment]
Its now my understanding that this is expected behavior for the AGIC
